File: tests/output/jpa/src/main/javagen/restaurant/jpa_feign/dtos/restaurant/PaiementItem.java
////
//// ATTENTION CE FICHIER EST GENERE AUTOMATIQUEMENT !
////

package restaurant.jpa_feign.dtos.restaurant;

import java.math.BigDecimal;
import java.time.LocalDateTime;

import jakarta.annotation.Generated;
import jakarta.validation.constraints.NotNull;

/**
 * Détail d'un paiement.
 */
@Generated("TopModel : https://github.com/klee-contrib/topmodel")
public record PaiementItem(
	@NotNull Integer factureId,
	@NotNull Integer swileCardId,
	@NotNull LocalDateTime dateCommande,
	@NotNull BigDecimal montantTotal
) {

	/**
	 * Enumération des champs de la classe {@link restaurant.jpa_feign.dtos.restaurant.PaiementItem PaiementItem}.
	 */
	public enum Fields {
		FACTURE_ID(Integer.class),
		SWILE_CARD_ID(Integer.class),
		DATE_COMMANDE(LocalDateTime.class),
		MONTANT_TOTAL(BigDecimal.class);

		private final Class<?> type;

		Fields(Class<?> type) {
			this.type = type;
		}

		/**
		 * Getter for type.
		 *
		 * @return value of {@link #type type}.
		 */
		public Class<?> getType() {
			return this.type;
		}
	}
}

File: tests/output/angular/src/appgenerated/model/restaurant/paiement-item.ts
////
//// ATTENTION CE FICHIER EST GENERE AUTOMATIQUEMENT !
////

import {DO_DATE_HEURE, DO_ID, DO_PRIX} from "@/domains";
import {e, entity, EntityToType} from "@focus4/entities";

export type PaiementItem = EntityToType<PaiementItemEntityType>;
export type PaiementItemEntityType = typeof PaiementItemEntity;

export const PaiementItemEntity = entity({
    factureId: e.field(DO_ID, f => f
        .label("restaurant.paiement.factureId")
        .comment("comments.restaurant.paiement.factureId")
    ),
    swileCardId: e.field(DO_ID, f => f
        .label("restaurant.paiement.swileCardId")
        .comment("comments.restaurant.paiement.swileCardId")
    ),
    dateCommande: e.field(DO_DATE_HEURE, f => f
        .label("restaurant.commande.dateCommande")
        .comment("comments.restaurant.commande.dateCommande")
    ),
    montantTotal: e.field(DO_PRIX, f => f
        .label("restaurant.commande.montantTotal")
        .comment("comments.restaurant.commande.montantTotal")
    )
});
